app/models/location.rb

```diff
@@ -1,16 +1,57 @@
 class Location < ApplicationRecord
-  # Search request to Location IQ API
-  def self.geosearch(query)
-    raise ArgumentError, "No query for geosearch" if query.blank?
-    raise "No API key for geosearch" unless ENV['LOCATION_IQ_API_KEY']
+  include LocationHelper
 
-    base_url = "https://us1.locationiq.com/v1/search.php?key=#{ENV['LOCATION_IQ_API_KEY']}&format=json&addressdetails=1&normalizecity=1"
-    query_url = "#{base_url}&q=#{query}"
+  # Base request to LocationIQ API
+  def self.location_api_request(endpoint_query)
+    raise "No location API key" unless ENV["LOCATION_IQ_API_KEY"]
+
+    query_url = "https://us1.locationiq.com/v1/#{endpoint_query}&key=#{ENV['LOCATION_IQ_API_KEY']}&format=json"
     uri = URI.parse(query_url)
     request = Net::HTTP::Get.new(uri)
-    resp = Net::HTTP.start(uri.host, uri.port, use_ssl: uri.scheme == 'https') do |http|
+    resp = Net::HTTP.start(uri.host, uri.port, use_ssl: true) do |http|
       http.request(request)
     end
     [resp.is_a?(Net::HTTPSuccess), JSON.parse(resp.body)]
   end
+
+  # Search request to Location IQ API
+  def self.geosearch(query)
+    raise ArgumentError, "No query for geosearch" if query.blank?
+    endpoint_query = "search.php?addressdetails=1&normalizecity=1&q=#{query}"
+    location_api_request(endpoint_query)
+  end
+
+  # Create a Location from parameters
+  def self.create_from_params(location_params)
+    # Ignore fields that don't match columns
+    location_params = location_params.select { |x| Location.attribute_names.index(x.to_s) }
+    # Name sanitization is imperfect but we can have a sanity check
```

Comment:
What does this mean?

Reply:
@tfrcarvalho I felt compelled to have some kind of sanitization but maybe it's not necessary..